### ISSUE TITLE
Use newtype instead of data says that the change decreases laziness in the presence of -XStrictData

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#285]
Using `newtype` over `data` shouldn't make a difference to the strictness when `-XStrictData` is active.

```
src/Foo.hs:73:1: Suggestion: Use newtype instead of data
      Found:
        data Bus = Bus{unBus :: Signed 16}
      Why not:
        newtype Bus = Bus{unBus :: Signed 16}
      Note: decreases laziness
```

**Comment by ndmitchell** [comment]
Hmm, I wonder if we need the error for things like:

    data T = C !a

Is that `data` really equivalent to `newtype` in all cases? Since `StrictData` is just that by default, it makes sense to not warn on bang patterns at the same time (if it really is safe, and I don't really know).

**Comment by expipiplus1** [comment]
It's not equivalent in all cases. I believe it's exactly the same if the data is unboxed though.

**Comment by ndmitchell** [comment]
I believe the StrictData just does the addition of bang patterns, or at least that's what the page at https://ghc.haskell.org/trac/ghc/wiki/StrictPragma says. In that case the warning might still be valid?

**Comment by expipiplus1** [comment]
I think it's equivalent in terms of strictness. Without unboxing it'll still put another indirection in there (which `newtype` won't).

**Comment by ndmitchell** [comment]
In that case the rule should be `data T = C !t` doesn't warn about strictness, and if StrictData is active that doesn't warn either.

**Comment by expipiplus1** [comment]
Yeah, sounds good to me.
